drivers/ethernet/dsa_ksz8863.h

```diff
@@ -8,8 +8,11 @@
 #define __DSA_KSZ8863_H__
 
 /* SPI commands */
-#define KSZ8863_SPI_CMD_WR (BIT(6))
-#define KSZ8863_SPI_CMD_RD (BIT(6) | BIT(5))
+//#define KSZ8863_SPI_CMD_WR (BIT(6))
+//#define KSZ8863_SPI_CMD_RD (BIT(6) | BIT(5))
+
+#define KSZ8863_SPI_CMD_WR (BIT(1))
+#define KSZ8863_SPI_CMD_RD (BIT(1) | BIT(0))
```

Comment:
KSZ8863 was used with bit-bang SMI driver and not SPI but that code is not checked in. I kept the SPI support as-is but wasn't able to test it since my board didn't have SPI connection to the PHY.

Reply:
@arvinf Have I understand correctly that the SPI part of KSZ8863 was not tested at all (and just copied from KSZ8794)?

Reply:
@lmajewski The root cause is described here: https://github.com/zephyrproject-rtos/zephyr/issues/42524 
It looks like, that KSZ8863 driver has not been test on Zephyr. It seems that KSZ8796 respect specification, but I cannot test it here, I do not have such HW.

Basically, KSZ8863 use different SPI protocol address is 8bit vs 16bit and positions of control bits are also different.

Reply:
@arvinf @paveldedourek-eaton @carlescufi 
- The SPI code for `KSZ8863` has not been tested up till @paveldedourek-eaton rework - I was just a copy & paste from KSZ8794
- The KSZ8863 datasheet:
https://ww1.microchip.com/downloads/en/DeviceDoc/KSZ8863MLL-FLL-RLL-Data-Sheet-00002335C.pdf
Figure 3-9 and Figure 3-10 shows that the addressing scheme and bit order is the same as for KSZ8794 (proper link and document reference shared in other reply in this thread).
Considering the above - the code for KSZ8794 shall work as well for KSZ8863.

Or am I missing something?

Reply:
@paveldedourek-eaton - Could you reply to the above comment? From what I can read from documentation of those two chips - there shall be no need for changing the SPI code for transmission to KSZ8863. Could you look to pointed out documentation and double check your setup? (Or correct me if I'm wrong)?

Reply:
For the record - I've pasted the PM you sent me (as you cannot add it to the github):
`SPI transaction contains in both case 3 bytes, with different content:`
`KSZ8794:`
`byte0 - 6bit+5bit -> COMMAND, 4..0bits -> address A11..A7`
`byte1 - 7..1bits -> address A6..A0, 0bit -> TR`
`byte2 - data`
`KSZ8863:`
`byte0 - 1bit+0bit -> COMMAND`
`byte1 - 7..0bits -> address A7..A0`
`byte2 - data`

As we are now 100% sure that the problem is not with LSB/MSB ordering, then IMHO we shall pull your changes.

(I'm _really_ surprised about the inconsistency about SPI transmission command byte format for both KSZ chips).